src/main/scala/org/ergoplatform/nodeView/state/DigestState.scala

```diff
@@ -158,19 +159,23 @@ object DigestState {
 
     (versionOpt, rootHashOpt) match {
       case (Some(version), Some(rootHash)) =>
-        val state = if (store.lastVersionID.isDefined && store.lastVersionID.forall(_.data sameElements version)) {
+        val state = if (store.lastVersionID.isDefined &&
+          store.lastVersionID.forall(w => bytesToVersion(w.data) == version)) {
```

Comment:
Line `store.lastVersionID.isDefined && store.lastVersionID.forall(w => bytesToVersion(w.data) == version)` could be replace with ` store.lastVersionID.map(w => bytesToVersion(w.data)).contains(version)`.

Also we could introduce helper method `def bawToVersion(w: BAW) = bytesToVersion(w.data)`.